misc/validate

```diff
@@ -2,30 +2,37 @@
 use 5.12.1;
 use warnings;
 
-use Dir::Self;
 use Getopt::Long;
 use JSON::Validator;
 use Mojo::File qw(path);
 use Mojo::JSON qw(decode_json);
 use Pod::Usage;
 
-my $schema_file = __DIR__ . '/../json-schema/input.yaml';
-my $name        = 'DeviceReport';
+my $schema_url = 'https://conch.joyent.us/schema/request';
+my $name       = 'DeviceReport';
 
 GetOptions(
-	'file|f:s'   => \$schema_file,
-	'schema|s:s' => \$name,
-	'report|r:s' => \my $report,
-	'help|h'     => \my $help,
+	'base_url|u:s'    => \$schema_url,
+	'schema_file|f:s' => \my $schema_file,
+	'schema|s:s'      => \$name,
+	'report|r:s'      => \my $report,
+	'help|h'          => \my $help,
 );
 
 pod2usage(1) if $help;
 
 my $validator = JSON::Validator->new;
-$validator->schema($schema_file);
+
+if ($schema_file){
+    my $schema = path($schema_file)->to_abs;
+    $validator->load_and_validate_schema($schema);
+} else {
+    $validator->load_and_validate_schema("$schema_url/$name");
+}
+
 $validator->coerce(1);
 
-$report = defined $report ? path($report)->slurp : do { local $/; <> };
+$report = decode_json defined $report ? path($report)->slurp : do { local $/; <> };
```

Comment:
need moar parentheses? internal keywords have very high operator precedence and everything else has very low, so I *think* what you want is `decode_json(defined $report ? path($report)->slurp : do { local $/; <> });` ?

Reply:
added moar parens.